roles/pre-flight-checks-openstack/tasks/prerequisites-sriov.yml

```diff
@@ -0,0 +1,59 @@
+---
+# There are prerequisites for SR-IOV support, including IOMMU enabled,
+# passthrough, and driver support. Many drivers support SR-IOV, but some
+# require out-of-kernel drivers (ie. Mellanox) and some don't support sysfs.
+
+- name: Fail on incompatible CPU architecture
+  fail:
+    msg: "Detected {{ cpu_vendor }} CPU not supported! Must be {{ supported_cpus }}."
+  when:
+    - cpu_vendor not in supported_cpus
+
+- name: Register IOMMU DMARs
+  find: paths=/sys/class/iommu file_type=directory patterns="*"
+  register: iommus
+  
+# We need to
+# - Check to see if IOMMU is already enabled. If not, let's check grub (and break out iommu/pt)
+
+- name: Check GRUB defaults and enable IOMMU if necessary
+  lineinfile:
+    path: /etc/default/grub
+    backrefs: true
+    regexp: '^GRUB_CMDLINE_LINUX="((?!.*{{ iommu_kernel_cmds }}).*)"$'
+    line: 'GRUB_CMDLINE_LINUX="\1 {{ iommu_kernel_cmds }}"'
+    backup: yes
+  register: grub
+
+- name: Update GRUB config (Ubuntu)
+  command: update-grub
+  when:
+    - ansible_distribution == 'Ubuntu'
+
+- name: Update GRUB config (CentOS)
+  command: grub2-mkconfig -o /boot/grub2/grub.cfg
+  when:
+    - ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
```

Comment:
This can be simplified to `when: ansible_os_family == "RedHat"` to align with other Express roles

Reply:
fixed.